nvflare/client/task_registry.py

```diff
@@ -114,6 +114,6 @@ def clear(self) -> None:
     def __str__(self):
         return f"{self.__class__.__name__}(config: {self.config.get_config()})"
 
-    def __del__(self):
+    def shutdown(self):
```

Comment:
Why does task registry have flare_agent? This design seems very odd. It is created when init the API, hence it should belong to the API, and shutdown there too.

Reply:
make sense, changed as suggested